ocaml/tests/test_auth_cache.ml

```diff
@@ -0,0 +1,178 @@
+(*
+ * Copyright (C) Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+module User = String
+
+module Secret = struct
+  type key = string
+
+  type digest = string
+
+  type salt = string
+
+  type secret = string
+
+  type t = digest * salt * secret
+
+  let create digest salt secret = (digest, salt, secret)
+
+  let read t = t
+
+  let create_salt () = Int32.(to_string (Random.int32 max_int))
```

Comment:
This now uses Random.int32 instead of fortuna, am I looking at the wrong code?

Reply:
Looks like github wasn't rendering the big diff by default, expand it shows the code.